src/univariate/continuous/cosine.jl

```diff
@@ -65,12 +65,24 @@ function logpdf{T<:Real}(d::Cosine{T}, x::Real)
     insupport(d, x) ? log(pdf(d, x)) : -T(Inf)
 end
 
-function cdf(d::Cosine, x::Real)
+function cdf{T<:Real}(d::Cosine{T}, x::Real)
+    if x < d.μ - d.σ 
+        return zero(T)
+    end
+    if x > d.μ + d.σ 
+        return one(T)
+    end
     z = (x - d.μ) / d.σ
     (1 + z + sinpi(z) * invπ) / 2
 end
 
-function ccdf(d::Cosine, x::Real)
+function ccdf{T<:Real}(d::Cosine, x::Real)
+    if x < d.μ - d.σ 
+        return one(T)
+    end
+    if x > d.μ + d.σ 
+        return zero(T)
```

Comment:
"Type stable" means that the type of the value returned from a function does not depend on the particular _value_ passed to it. As a contrived example, `f(x::Int) = x > 1 ? x : 1.0` is not type stable, since the result is `Float64` when the input is less than 2 and `Int` otherwise. This matters because Julia can get a lot of speedup from knowing or being able to predict the type of an object, plus it's less surprising for a user when the return type is consistent across input values of a particular type.

When values of different types interact through some operation, their types undergo "promotion," which means that the values are converted to a common type before the operation is carried out. There's a hierarchy that determines the exact mapping from combinations of types to a common type, but in general it's pretty intuitive. An example of promotion is `1 + 1.0`, which is `Int + Float64 -> Float64`.

In situations such as this, one can use the `promote` function or one of its kin to figure out what the return type should be. That's what I was referring to with my "`promote` magic" comment.

In this particular case I don't think we need to worry to much about getting the return types perfect, since as you pointed out, this style is used for other functions as well. Perhaps someone should eventually go through and try to make these things type stable, but don't worry about that now.